src/api/ApiClient.ts

```diff
@@ -0,0 +1,77 @@
+import Axios from 'axios';
+import type { AxiosInstance } from 'axios';
+
+export type HttpHeaders = {
+  [key: string]: string;
+};
+
+export type RequestConfig = {
+  headers: HttpHeaders;
+};
+
+export class ApiConfiguration {
+  accessToken?: string;
+}
+
+export interface IApiClient {
+  post<Request, Response>(path: string, object: Request, config?: RequestConfig): Promise<Response>;
+  patch<Request, Response>(path: string, object: Request): Promise<Response>;
+  put<Request, Response>(path: string, object: Request): Promise<Response>;
+  get<Response>(path: string): Promise<Response>;
+}
+
+export default class ApiClient implements IApiClient {
+  private client: AxiosInstance;
+
+  protected createAxiosClient(): AxiosInstance {
+    return Axios.create({
+      baseURL: 'http://localhost:3000',
+      responseType: 'json' as const,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      timeout: 10 * 1000,
```

Comment:
What do these magic numbers mean?

Reply:
It represents the number of milliseconds that need to be set for the request to time out. This represents milliseconds; I'll add this to a variable called milliseconds.

Reply:
That's a good start, as a rule of thumb if you're not using the default, adding a breadcrumb comment on the PR and sometimes even in code can be useful